src/vehicles/use-vehicles.ts

```diff
@@ -0,0 +1,37 @@
+import {useEffect, useState} from 'react';
+import {FeatureCollection, GeoJSON} from 'geojson';
+import {VehicleFragment} from '@atb/api/types/generated/fragments/vehicles';
+import {toFeatureCollection, toGeoJSONFeature} from '@atb/components/map/utils';
+import {Coordinates} from '@atb/utils/coordinates';
+import {getVehicles} from '@atb/api/vehicles';
+import {useIsVehiclesEnabled} from '@atb/vehicles/use-vehicles-enabled';
+
+export const useVehicles = () => {
+  const [lat, setLat] = useState(0);
+  const [lon, setLon] = useState(0);
+  const [range, setRange] = useState(500);
+  const [vehicles, setVehicles] = useState<
+    FeatureCollection<GeoJSON.Point, VehicleFragment>
+  >(toFeatureCollection([]));
+  const isVehiclesEnabled = useIsVehiclesEnabled();
+
+  useEffect(() => {
+    if (isVehiclesEnabled) {
+      getVehicles({lat, lon, range})
+        .then(toGeoJSONFeature)
+        .then(toFeatureCollection)
+        .then(setVehicles);
+    }
+  }, [lon, lat, range, isVehiclesEnabled]);
```

Comment:
We probably need to do something with `AbortController` and the `useEffect` cleanup function, if not we risk race conditions. Like if you change regions quite fast, and the api response from the first region change returns after the api response from the second region, then the vehicles from the first region change will overwrite the vehicles from the second region change.

Reply:
Yes, I suppose you are right. I'll see what I can come up with

Reply:
Har lagt inn et forslag nå, men det er litt utfordrende å teste om det fungerer som forventet.